cmd/git-remote-ipld/ipld.go

```diff
@@ -242,16 +242,41 @@ func (h *IpnsHandler) Push(remote *core.Remote, local string, remoteRef string)
 func (h *IpnsHandler) bigNodePatcher(tracker *core.Tracker) func(cid.Cid, []byte) error {
 	return func(hash cid.Cid, data []byte) error {
 		if len(data) > (1 << 21) {
-			c, err := h.api.Add(bytes.NewReader(data))
+			endOfHeader := -1
+			for i := 0; i < len(data); i++ {
+				if data[i] == 0 {
+					endOfHeader = i + 1
+					break
+				}
+			}
+
+			var objectHash string
+			var err error
+			if endOfHeader > 0 {
+				headerHash, err := h.api.Add(bytes.NewReader(data[0:endOfHeader]))
+				if err != nil {
+					return err
+				}
+
+				dataHash, err := h.api.Add(bytes.NewReader(data[endOfHeader:]))
```

Comment:
These two adds can run in parallel with `sync.waitgroup` (api calls can be slow, but can be heavily parallelized)